contracts/locker/BaseLocker.sol

```diff
@@ -75,6 +83,47 @@ contract BaseLocker is
         _setApprovalForAll(address(this), _staking, true);
     }
 
+    /**
+     * @notice Sets the underlying token address
+     * @param _token The new token address
+     */
+    function setTokenAddress(address _token) external onlyOwner {
+        address oldToken = address(underlying);
+        underlying = IERC20(_token);
+        emit TokenAddressSet(oldToken, _token);
+    }
+
+    /**
+     * @notice Sets the staking contract address
+     * @param _staking The new staking contract address
+     */
+    function setStakingAddress(address _staking) external onlyOwner{
+        address oldStaking = address(staking);
+        staking = IOmnichainStaking(_staking);
+        emit StakingAddressSet(oldStaking, _staking);
+    }
+
+    /**
+     * @notice Sets the staking bonus contract address
+     * @param _stakingBonus The new staking bonus contract address
+     */
+    function setStakingBonusAddress(address _stakingBonus) external onlyOwner {
+        address oldStakingBonus = address(0); // Update this with the actual old staking bonus address if needed
```

Comment:
Ideally you can already do that using the value from `_stakingBonus` instead of using `address(0)`

Reply:
Done https://github.com/zerolend/governance/pull/35/commits/a1e276769034c0bb0928056f8320418eedb05851